arduino-core/src/processing/app/Sketch.java

```diff
@@ -355,8 +355,8 @@ public void saveAs(File newFolder) throws IOException {
     // Copy the data folder (this may take a while.. add progress bar?)
     if (getDataFolder().exists()) {
       File newDataFolder = new File(newFolder, "data");
-      // Create the data folder
-      if (!newDataFolder.mkdirs()) {
+      // Check if data folder exits, if not try to create the data folder
+      if (!(newDataFolder.exists() || newDataFolder.mkdirs())) {
```

Comment:
For readability you might even want to separate the condition into two if statements. (In case you don't know how the lazy and/or operator works).

But I think it is better to keep them into one if statement. I will change it to the and operator for better readability.